dev_docs/trace_metrics.md

```diff
@@ -0,0 +1,72 @@
+# Trace metrics
+
+APM Server aggregates several types of metrics from trace events: transaction
+metrics, and service destination metrics. This process is sometimes referred
+to as "pre-aggregation": aggregation of events prior to indexing, and indexing
+the aggregated metrics, as opposed to aggregation of documents in Elasticsearch.
+
+## Transaction metrics
+
+Transaction metrics measure the latency distribution for transaction groups.
+As transactions are observed by APM Server, it groups them according to various
+attributes such as `service.name`, `transaction.name`, and `kubernetes.pod.name`.
+The latency is then recorded in an [HDRHistogram](http://hdrhistogram.org/) for
+that group. Transaction group latency histograms are periodically indexed (every
+minute by default), with configurable precision (defaults to 2 significant figures).
+
+To protect against memory exhaustion due to high-cardinality transaction names
+(or other attributes), APM Server places a limit on the number of transaction
+groups that are tracked at any given time. By default, this limit is 10000
+transaction groups. When transaction group latency histograms are indexed, the
+groups are reset, enabling a different set of 10000 groups to be recorded.
+
+## Service destination metrics
+
+Service destination metrics measure throughput and average latency of operations
+from one service to another. This works much the same as transaction metrics
+aggregation: span events describing an operation that involves another service
+are grouped by the originating and target services, and the span latency is
+accumulated. For these metrics we record only a count and sum, enabling calculation
+of throughput and average latency. Once again, a default limit of 10000 groups is
+imposed.
+
+## Dealing with sampling
+
+The APM app in Kibana visualises transaction throughput, both as an overall
+throughput and as a histogram of transaction counts bucketed by latency. In the
+past we powered these metrics by recording a document in Elasticsearch for every
+transaction, sampled or not. We now use transaction metrics. The specific details
+of the queries are documented at
+https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/dev_docs/apm_queries.md
+
+Transaction metrics are aggregated from transaction events. To avoid agents
+having to send events to APM Server for non-sampled traces, we instead have
+agents include the sampling rate in events that it does send. APM Server then
+multiplies metrics by the inverse of the sampling rate. For example, if a
```

Comment:
This might be overly pedantic, but adding a short explanation as to why we're multiplying the inverse might be helpful for some people, eg. "assuming the sampled traces comprise a representative sample, apm server multiplies by the inverse of the sampling rate so that recorded metrics are scaled to include both sampled and unsampled txs" or something

Reply:
Good call. I added some more words, please let me know if it's any clearer.